lib/msf/ui/console/driver.rb

```diff
@@ -504,6 +504,15 @@ def unknown_command(method, line)
         end
         self.busy = false
         return
+      elsif framework.modules.create(method)
+        super
+        print_error("This is a module we can load. Do you want to use #{method}?")
```

Comment:
Looks like `prompt_yesno` isn't available in this context. Could you change this interaction to match that? I'd like the user experience to be consistent.

Also, if you interrupt the input line, the prompt doesn't get restored right away:

```
msf5 > exploit/windows/smb/ms17_010_eternalblue
[-] Unknown command: exploit/windows/smb/ms17_010_eternalblue.
[-] This is a module we can load. Do you want to use exploit/windows/smb/ms17_010_eternalblue?
(y/N) > Interrupt: use the 'exit' command to quit
(y/N) > N
[-] Unknown command: N.
(y/N) > y
[-] Unknown command: y.
(y/N) > exploit/windows/smb/ms17_010_eternalblue
[-] Unknown command: exploit/windows/smb/ms17_010_eternalblue.
[-] This is a module we can load. Do you want to use exploit/windows/smb/ms17_010_eternalblue?
(y/N) > y
msf5 exploit(windows/smb/ms17_010_eternalblue) > back
msf5 >
```

Sadly, due to how we implement prompts, it's kinda fair game what gets to change it.

Reply:
Ah, all good points @wvu-r7. Fixed, and refactored into a method in `shell.rb` for future use due to the gotchas.